packages/react-components/src/components/RemoteVideoTile.tsx

```diff
@@ -99,22 +105,67 @@ export const _RemoteVideoTile = React.memo(
       );
     }, [renderElement, isReceiving]);
 
+    const showLabelTrampoline = useMemo(() => {
+      /* @conditional-compile-remove(one-to-n-calling) */ /* @conditional-compile-remove(PSTN-calls) */
+      return canShowLabel(props.participantState, props.isNarrow, props.showLabel, containerWidth);
+      return props.showLabel;
+    }, [
+      /* @conditional-compile-remove(one-to-n-calling) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      containerWidth,
+      props
+    ]);
+
     return (
-      <VideoTile
-        key={userId}
-        userId={userId}
-        renderElement={renderVideoStreamElement}
-        displayName={displayName}
-        onRenderPlaceholder={onRenderAvatar}
-        isMuted={isMuted}
-        isSpeaking={isSpeaking}
-        showMuteIndicator={showMuteIndicator}
-        showLabel={props.showLabel}
-        personaMinSize={props.personaMinSize}
-        /* @conditional-compile-remove(one-to-n-calling) */
-        /* @conditional-compile-remove(PSTN-calls) */
-        participantState={participantState}
-      />
+      <div ref={containerRef}>
+        <VideoTile
+          key={userId}
+          userId={userId}
+          renderElement={renderVideoStreamElement}
+          displayName={displayName}
+          onRenderPlaceholder={onRenderAvatar}
+          isMuted={isMuted}
+          isSpeaking={isSpeaking}
+          showMuteIndicator={showMuteIndicator}
+          personaMinSize={props.personaMinSize}
+          showLabel={showLabelTrampoline}
+          /* @conditional-compile-remove(one-to-n-calling) */
+          /* @conditional-compile-remove(PSTN-calls) */
+          participantState={props.participantState}
+        />
+      </div>
     );
   }
 );
+
+/* @conditional-compile-remove(one-to-n-calling) */
+/* @conditional-compile-remove(PSTN-calls) */
+/**
+ * Determines if a label should be shown for a remote video tile.
+ * When the remote video tile is rendered as a small tile in horizontal gallery,
+ * we hide the participants name if they are in hold/connecting states.
+ */
+const canShowLabel = (
+  participantState?: ParticipantState,
+  isNarrow?: boolean,
+  showLabel?: boolean,
+  containerWidth?: number
+): boolean | undefined => {
+  const isCallingOrHold = (participantState?: ParticipantState): boolean => {
```

Comment:
nit: this can be defined outside this function
(this type of function is a good candidate for a unit test that can catch an error when a new participant state is added and this function may or may not need updating)